docs/book/src/clusteropenstack/configuration.md

```diff
@@ -177,6 +178,57 @@ during an upgrade. When the API server has a floating IP, this role is fulfilled
 floating IP even if there is no load balancer. When the API server does not have a floating
 IP, the load balancer virtual IP on the cluster network is used.
 
+## Restrict Access to the API server
+
+It is possible to restrict access to the Kubernetes API server on a network level. If required, you can specify
+the allowed CIDRs by `spec.APIServerLoadBalancer.AllowedCIDRs` of `OpenStackCluster`.
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha5
+kind: OpenStackCluster
+metadata:
+  name: <cluster-name>
+  namespace: <cluster-namespace>
+spec:
+  allowAllInClusterTraffic: true
+  apiServerLoadBalancer:
+    allowedCidrs:
+    - 192.168.10/24
+    - 10.10.0.0/16
+```
+
+All known IPs of the target cluster will be discovered dynamically (e.g. you don't have to take care of target Cluster own Router IP, internal CIDRs or any Bastion Host IP).
+**Note**: Please ensure, that at least the outgoing IP of your management Cluster is added to the list of allowed CIDRs. Otherwise CAPO can't reconcile the target Cluster correctly.
```

Comment:
or I saw the code seems the IP is added in the code directly?

Reply:
but how do we know the router IP of the management cluster? 
There might be so many different CAPO-deployment models (i guess) where it's not that easy to determine the outgoing IP of the management cluster (some of my thoughts where i don't had an easy answer to myself :-) ):
* Management Cluster might run somewhere else - how to determine the IP
* OpenstackCredentials during cluster provisioning are mostly scoped to target openstack tenant:
   even if the management cluster is self-managed by the same controller, how to tell in a generic way during reconciliation what's the `openstackcluster` object of the management cluster (by setting an annotation to the `openstackcluster` object of the management cluster?)
   
   for sure, it might be possible to implement something, my idea was to not implement to much "magic" for now and keep the feature as simple as possible. But i aggree with you, without setting the proper managmentcluster router ip, CAPO users will feel the pain. But before we start implementing this as feature, i guess we should discuss how the management-cluster-discovery could look like. If i can share/initiate the discussion.

Reply:
yes, so we can follow  that pattern and implement here, then doc the requirements as well :)